### PR TITLE
Allow zooming in to the current location when toggling the lock 2nd time

### LINKED ISSUE #1956: [Bug] Location lock is not zooming in when activating it for 2nd time

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #1956

@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt [hunk 1]
```diff
@@ -72,6 +72,9 @@ constructor(
   @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
 ) : AbstractViewModel() {
 
+  /** This is set to true as soon as the location lock is disabled. */
+  private var isLocationLockReset = true
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:76]: In that case should this be named for the positive case, `locationLockEnabled`?

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt [hunk 2]
```diff
@@ -206,9 +210,9 @@ constructor(
   private suspend fun updateCameraPositionOnLocationChange() {
     getLocationUpdates()
       .map { it.toCoordinates() }
-      .withIndex()
-      .collect { (index, coordinates) ->
-        if (index == 0) {
+      .collect { coordinates ->
+        if (isLocationLockReset) {
+          isLocationLockReset = false
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:215]: The way it's written now kit's not immediately clear what we're trying to do something different on the first emission. Would it be more idiomatic to still use `withIndex()` and to reset the stream each time to location updates are enabled (maybe via flatMapLatest?)

**Review by gino-m — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:76]: The intention is to mark the field as false as soon as we consume it. `locationLockEnabled` would mean that we can't set the bit to false after it's first use.

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt (hunk 2 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:215]: I wasn't sure how to reset the stream. Will check once again.

Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:215]:
```kt
  locationLockEnabled
    .flatMapLatest { enabled -> getLocationUpdates().filter { enabled } }
    .withIndex()
    .collect { ...
    }
```

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt (hunk 1 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:76]: I see. In any case I think the functional approach described below is easier to understand than managing an additional state and relying on side-effects.

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt]: This is now reverted

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt (hunk 2 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt]: Thanks for the suggestion! This also gets rid of the boolean bit

**Review by gino-m — APPROVED** [review]
🚀 !!!

**Pull request merged**